### ISSUE TITLE
Error when using non-english locale on Debian

### ISSUE THREAD

**Issue by lilalkor** [certbot/certbot#3876]
Using ru_RU locale leads to error:

```
root@localhost /tmp/cbot # ./certbot-auto
Bootstrapping dependencies for Debian-based OSes...
В кэше http://security.debian.org wheezy/updates Release.gpg
В кэше http://security.debian.org wheezy/updates Release                                                                                                                                               
В кэше http://ftp.debian.org wheezy-backports Release.gpg                                                                                                                                              
В кэше http://security.debian.org wheezy/updates/main Sources                                                                                                      
В кэше http://ftp.debian.org wheezy-backports Release                                                                                                                                                   
В кэше http://security.debian.org wheezy/updates/contrib Sources                                                                                                                      
В кэше http://security.debian.org wheezy/updates/non-free Sources                                                                                                                            
В кэше http://security.debian.org wheezy/updates/main amd64 Packages                                                                                                                                                                   
В кэше http://security.debian.org wheezy/updates/contrib amd64 Packages                                                                                                                                                                
В кэше http://security.debian.org wheezy/updates/non-free amd64 Packages                                                                                                                                                               
В кэше http://ftp.debian.org wheezy-backports/main Sources                                                                                                                                              
В кэше http://http.debian.net wheezy Release.gpg                                                                                                                
В кэше http://ftp.debian.org wheezy-backports/contrib Sources                                                                                                                                                                        
В кэше http://http.debian.net wheezy-updates Release.gpg                                                                                                                                                  
В кэше http://ftp.debian.org wheezy-backports/non-free Sources                                                                                                                                            
В кэше http://http.debian.net wheezy-backports Release.gpg                                                                                                                                                
В кэше http://ftp.debian.org wheezy-backports/main amd64 Packages                                                                                                 
В кэше http://http.debian.net wheezy-backports-sloppy Release.gpg                                                                                                 
В кэше http://ftp.debian.org wheezy-backports/contrib amd64 Packages                                                                                              
В кэше http://ftp.debian.org wheezy-backports/non-free amd64 Packages                                                                                             
В кэше http://http.debian.net wheezy Release                                                                                                                                                   
Получено:1 http://http.debian.net wheezy-updates Release [155 kB]                                                                                                                              
В кэше http://http.debian.net wheezy-backports Release                                                                                                                        
В кэше http://http.debian.net wheezy-backports-sloppy Release                                                                                                                                    
В кэше http://http.debian.net wheezy/main Sources                                                                                                 
В кэше http://http.debian.net wheezy/contrib Sources                                                                        
В кэше http://http.debian.net wheezy/non-free Sources                                                 
В кэше http://nginx.org wheezy Release.gpg                                                            
В кэше http://http.debian.net wheezy/main amd64 Packages                                              
Получено:2 http://http.debian.net wheezy/contrib amd64 Packages [42,0 kB]                             
Получено:3 http://http.debian.net wheezy/non-free amd64 Packages [80,8 kB]                                                             
В кэше http://http.debian.net wheezy-backports/main Sources                                                                           
В кэше http://http.debian.net wheezy-backports/contrib Sources                                                                                
В кэше http://nginx.org wheezy Release                                                                                                        
Получено:4 http://http.debian.net wheezy-backports/non-free Sources [8 935 B]                                                 
В кэше http://http.debian.net wheezy-backports/main amd64 Packages                                                                      
В кэше http://http.debian.net wheezy-backports/contrib amd64 Packages                                                                        
В кэше http://http.debian.net wheezy-backports/non-free amd64 Packages                                                                       
В кэше http://nginx.org wheezy/nginx Sources                                                           
Получено:5 http://http.debian.net wheezy-updates/main Sources [5 505 B]                                
Получено:6 http://http.debian.net wheezy-updates/contrib Sources [14 B]                                                                   
Получено:7 http://http.debian.net wheezy-updates/non-free Sources [929 B]                                                         
Получено:8 http://http.debian.net wheezy-updates/main amd64 Packages [7 047 B]                                                      
Получено:9 http://http.debian.net wheezy-updates/contrib amd64 Packages [14 B]                                                             
Получено:10 http://http.debian.net wheezy-updates/non-free amd64 Packages [488 B]                                                  
В кэше http://nginx.org wheezy/nginx amd64 Packages                                                                                   
В кэше http://http.debian.net wheezy-backports-sloppy/main Sources                                                                    
В кэше http://http.debian.net wheezy-backports-sloppy/contrib Sources                                                                 
В кэше http://http.debian.net wheezy-backports-sloppy/non-free Sources
В кэше http://http.debian.net wheezy-backports-sloppy/main amd64 Packages
В кэше http://http.debian.net wheezy-backports-sloppy/contrib amd64 Packages                            
В кэше http://http.debian.net wheezy-backports-sloppy/non-free amd64 Packages                           
Получено 301 kБ за 1с (288 kБ/c)                                 
Чтение списков пакетов… Готово
Чтение списков пакетов… Готово
Построение дерева зависимостей       
Чтение информации о состоянии… Готово
Пакет virtualenv недоступен, но упомянут в списке зависимостей другого пакета.
Это может означать, что пакет отсутствует, устарел, или доступен из источников, не упомянутых в sources.list

E: Для пакета «virtualenv» не найден кандидат на установку
```
Forcing en_US locale works fine:
```
# LC_ALL=en_US.utf8 ./certbot-auto 
Bootstrapping dependencies for Debian-based OSes...
Hit http://security.debian.org wheezy/updates Release.gpg
Hit http://security.debian.org wheezy/updates Release                                                                                                                    
Hit http://ftp.debian.org wheezy-backports Release.gpg                                                                                                                   
Hit http://security.debian.org wheezy/updates/main Sources                                                           
Hit http://ftp.debian.org wheezy-backports Release                                                                                     
Hit http://security.debian.org wheezy/updates/contrib Sources                                                                          
Hit http://security.debian.org wheezy/updates/non-free Sources                                                                                                                    
Hit http://security.debian.org wheezy/updates/main amd64 Packages                                                                                                                 
Hit http://security.debian.org wheezy/updates/contrib amd64 Packages                                                                                                              
Hit http://security.debian.org wheezy/updates/non-free amd64 Packages                                                                                                             
Hit http://ftp.debian.org wheezy-backports/main Sources                                                                                
Hit http://http.debian.net wheezy Release.gpg                                                                                          
Hit http://ftp.debian.org wheezy-backports/contrib Sources                                                                                                                    
Hit http://ftp.debian.org wheezy-backports/non-free Sources                                                                                                                   
Hit http://http.debian.net wheezy-updates Release.gpg                                                                                              
Hit http://ftp.debian.org wheezy-backports/main amd64 Packages                                                                                                                            
Hit http://http.debian.net wheezy-backports Release.gpg                                                                                                                                   
Hit http://ftp.debian.org wheezy-backports/contrib amd64 Packages                                                                                           
Hit http://http.debian.net wheezy-backports-sloppy Release.gpg                                                                                                                 
Hit http://ftp.debian.org wheezy-backports/non-free amd64 Packages                                                                                                             
Hit http://http.debian.net wheezy Release                                                                         
Hit http://http.debian.net wheezy-updates Release                                         
Hit http://http.debian.net wheezy-backports Release                                       
Hit http://http.debian.net wheezy-backports-sloppy Release                                                               
Get:1 http://http.debian.net wheezy/main Sources [5,993 kB]                                                                  
Hit http://http.debian.net wheezy/contrib Sources                              
Hit http://nginx.org wheezy Release.gpg                                                         
Hit http://http.debian.net wheezy/non-free Sources                            
Hit http://http.debian.net wheezy/main amd64 Packages                                          
Hit http://http.debian.net wheezy/non-free amd64 Packages                                           
Hit http://http.debian.net wheezy-updates/main Sources                                              
Get:2 http://http.debian.net wheezy/contrib amd64 Packages [42.0 kB]                                           
Hit http://http.debian.net wheezy-updates/contrib Sources                                                                                       
Hit http://nginx.org wheezy Release                                                                            
Hit http://http.debian.net wheezy-updates/non-free Sources                                                                                             
Hit http://http.debian.net wheezy-updates/main amd64 Packages                                                                   
Hit http://http.debian.net wheezy-updates/contrib amd64 Packages                         
Hit http://http.debian.net wheezy-updates/non-free amd64 Packages                        
Hit http://nginx.org wheezy/nginx Sources                                                
Get:3 http://http.debian.net wheezy-backports/main Sources [453 kB]                      
Hit http://http.debian.net wheezy-backports/contrib Sources                                                           
Hit http://http.debian.net wheezy-backports/non-free Sources                                                           
Hit http://http.debian.net wheezy-backports/main amd64 Packages                                                       
Hit http://http.debian.net wheezy-backports/contrib amd64 Packages                                            
Hit http://http.debian.net wheezy-backports/non-free amd64 Packages                                            
Hit http://nginx.org wheezy/nginx amd64 Packages                                            
Hit http://http.debian.net wheezy-backports-sloppy/main Sources                          
Hit http://http.debian.net wheezy-backports-sloppy/contrib Sources    
Hit http://http.debian.net wheezy-backports-sloppy/non-free Sources
Hit http://http.debian.net wheezy-backports-sloppy/main amd64 Packages
Hit http://http.debian.net wheezy-backports-sloppy/contrib amd64 Packages
Hit http://http.debian.net wheezy-backports-sloppy/non-free amd64 Packages
Fetched 6,488 kB in 1s (3,969 kB/s)      
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
ca-certificates is already the newest version.
ca-certificates set to manually installed.
gcc is already the newest version.
python is already the newest version.
python set to manually installed.
python-dev is already the newest version.
python-virtualenv is already the newest version.
augeas-lenses is already the newest version.
libaugeas0 is already the newest version.
The following extra packages will be installed:
  libssl1.0.0
The following NEW packages will be installed:
  dialog libffi-dev
The following packages will be upgraded:
  libssl-dev libssl1.0.0
2 upgraded, 2 newly installed, 0 to remove and 143 not upgraded.
Need to get 3,514 kB of archives.
After this operation, 1,867 kB of additional disk space will be used.
Do you want to continue [Y/n]? ^C
```

The reason is packets check in line 180 that uses english error text:

```
  176   # virtualenv binary can be found in different packages depending on
  177   # distro version (#346)
  178 
  179   virtualenv=
  180   if apt-cache show virtualenv > /dev/null 2>&1 && ! apt-cache --quiet=0 show virtualenv 2>&1 | grep -q 'No packages found'; then
  181     virtualenv="virtualenv"
  182   fi
  183 
  184   if apt-cache show python-virtualenv > /dev/null 2>&1; then
  185     virtualenv="$virtualenv python-virtualenv"
  186   fi
```

The easiest way to fix it is just set locale in script to en_US.utf8 (or C, or anything you want).

**Comment by pde** [comment]
If someone wants to send a PR this should be fixed in `letsencrypt-auto-source/pieces/bootstrappers/deb_common.sh` (then run `letsencrypt-auto-source/build.py` and commit that set of changes).

**Comment by pde** [comment]
Actually, there are other places in the script that assume English scripts, so instead the locale should be set near the beginning of `letsencrypt-auto-source/letsencrypt-auto.template`.

We should think about what assumptions are made by setting the locale to various things, but `C` is probably comparatively safe?

**Comment by pde** [comment]
@lilalkor thanks for a great bug report!

**Comment by bmw** [comment]
Fixed in #3900.
